doc/design/figures/xgboost_feature_column.md

```diff
@@ -0,0 +1,131 @@
+# Feature Column Support in SQLFlow XGBoost models
+
+## Overall
+
+SQLFlow extends SQL grammar to support data pre-processing using `COLUMN` clauses. 
+For example, we can use `CATEGORY_HASH` to parse a string column to an integer column, which is usually a common data pre-processing operation in NLP tasks.
+
+```sql
+SELECT string_column1, int_column2, class FROM xgboost.gbtree
+TO TRAIN XGBoostModel
+COLUMN INDICATOR(CATEGORY_HASH(string_column1, 10)), int_column2
+LABEL class 
+INTO sqlflow_xgboost_model.my_model;
+```
+
+Currently, `COLUMN` clauses are supported in SQLFlow TensorFlow models. The `COLUMN` clauses are transformed
+into TensorFlow feature column API calls inside SQLFlow codegen implementation.
+
+However, XGBoost has no similar feature column APIs as TensorFlow. Currently, XGBoost models can only support simple column names like `c1, c2, c3` in `COLUMN` clauses,
+and any data pre-processing is not supported. It makes that we cannot use XGBoost to train NLP models which accept string column as their input.
```

Comment:
Not only NLP models will be supported but also models that takes categorical string feature inputs.

Reply:
Thanks. Done.